easybuild/framework/easyconfig/templates.py

```diff
@@ -216,9 +216,12 @@ def template_constant_dict(config, ignore=None, skip_lower=None, toolchain=None)
         # copy to avoid changing original list below
         deps = copy.copy(config.get('dependencies', []))
 
-        # only consider build dependencies for defining *ver and *shortver templates if we're in iterative mode
-        if hasattr(config, 'iterating') and config.iterating:
-            deps += config.get('builddependencies', [])
+        if hasattr(config, 'iterating'):
+            if 'builddependencies' in config.iterate_options:
+                if config.iterating:
+                    deps += config.get('builddependencies', [])
+            else:
+                deps += config.get('builddependencies', [])
```

Comment:
@ekouts I think we can simplify this whole block, there's no point in being overly careful I think...

The iterating part is now basically irrelevant, we can always consider build deps to define template values:

```suggestion
        deps += config.get('builddependencies', [])
```

Reply:
No, my suggestion is too simplistic... We need to take into account that `builddependencies` can be a list of lists while we're doing an iterative installation (like when `multi_deps` is used), but we also just want to add all builddependencies if we're not doing an iterated installation...